governance/delegation-interop.md

```diff
@@ -0,0 +1,65 @@
+
+# Summary
+
+There will be two main changes to the existing `GovernanceDelegation` contract to support interoperability. First, the function in which the checkpoints can be queried from the
+contract will now check for the migration status prior to retrieving voting power and implement an announce function that will emit an event that contains the chain ID and the block
+number of the latest checkpoint. The contract on OP mainnet will then consume these events, and in its own storage provide a mapping between chainIds and an object containing both
+the latest checkpoint and the block number. After tokens are transferred using the `GovToken` contract a check is performed on the delegation contract that ensures that before this
+mapping is updated, the nonce of the token transfer received has been incremented as well as being sent from a block number greater than the current one.
+
+# Problem Statement + Context
+
+In the existing `GovernanceDelegation` and `GovToken` contract it does not support interoperability, thus making the voting power fragmented across all of the potential L2 chains. The negative consequence of this is that tracking delegations becomes a much harder task when users have partially delegated to many different users that may or may not be on the same chain.
+
+However, pushing all of the logic onto the `GovToken` proves to be problematic given the potential flood of messages that may be out of sequence or have many smaller transfer events
+prior to a single delegation event. The following section aims to outline a solution to correct this as well as ensure that relayers still have the proper incentives to handle these
+messages.
+
+# Alternatives Considered
+
+When originally considering this problem, the idea was to create a hook during the `afterTokenTransfer` event. As alluded to above, this do not provide some of the desired functionality.It also makes a few assumptions about the behaviors of delegates that receive tokens and stifles their ability to create partial delegations in a low-cost manner. Moreover, if there was
+a want to later batch updates of voting power to mainnet it would be costly to those operating governance.
+
+Another active consideration is the use of block timestamp of each L2 instead of block number while implementing this new solution. The primary reason for block numbers to be used in
+its stead, stems from a concern of manipulation by sequencers and the possible delays that might need to be enforced between events to ensure correctness.
+
+# Proposed Solution
+
+## Rationale
+
+Due to the fact that the `GovernanceDelegation` contract now has awareness of the block numbers and chain IDs of all of the voting power checkpoints each L2 can now be responsible for
```

Comment:
Can you expand on this awareness a bit? I see some mention of it in the overview. It could be useful to have some simple data structures in here to make it easier to visualize.

Also i would like to avoid branching logic based on the chainid. It would be most ideal if the events could be imported into any chain but with no real incentive to import them anywhere besides op mainnet. This will simplify testing, remove edge cases and enable delegation power to be observed from any chain

Reply:
Was unsure if pseudocode was bad style for a design doc, but specifically I was considering some type of mapping in solidity between `mapping ( uint16 => Checkpoint ) ` where the data structure is `ERC20Votes.Checkpoint[]` contains an additional field `blockNumber`